### PR TITLE
added module-info.java

### LINKED ISSUE #1020: Datafaker as a module
It would be great if you can add a module-info.java and make it available from modular projects.

### PR DESCRIPTION
**Pull request by pbankovic**
Support for modular projects, fixes #1020

### CODE REVIEW

**Comment by bodiam** [comment]
I understand why you did it, but should we really expose the internal package?

**Comment by pbankovic** [comment]
Actually, we shouldn't. But after I saw that the build was failing, I thought not exposing the internal package caused it.
I'm gonna remove it again. 
Would yo say that net.datafaker.internal.helper is the only package that should stay internal, or is there anything else that shouldn't be referenced from outside?

**Comment by bodiam** [comment]
It seems the build is still failing. I wouldn't worry too much about exposing the helper package right now, we can address that later, but I think it's indeed the only package which ideally should stay hidden

**Comment by pbankovic** [comment]
Yes, I see that it fails to generate javadoc, but I just can't see how it could be affected by the module declaration file.

File: pom.xml [hunk 1]
```diff
@@ -362,6 +362,7 @@
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <version>${maven-javadoc-plugin.version}</version>
                 <configuration>
+					<sourcepath>src/main/java</sourcepath>
```
Comment (kingthorin) [inline pom.xml:365]: Could you indent this to be 'inside' configuration?

Comment (pbankovic) [reply pom.xml:365]: Oh, yes, I guess my IDE is configured differently

**Comment by pbankovic** [comment]
I had to add the source path to the javadoc plugin, and now it's fine.
Fixing the indention, and that should be it  😄

**Review by kingthorin — APPROVED** [review]

**Comment by bodiam** [comment]
I know, I know. I need to refresh that key. Happy if you want to merge it though

**Comment by kingthorin** [comment]
Okay, thanks!

**Pull request merged**